pkg/apis/eventing/v1/broker_lifecycle.go

```diff
@@ -31,6 +33,18 @@ var brokerCondSet = apis.NewLivingConditionSet(
 	BrokerConditionAddressable,
 )
 
+var customConditionSet = map[string]apis.ConditionSet{
```

Comment:
My personal preference would be that we start with a single conditionSet that the controller sets to reduce complexity. As far as I understand there are no plans to have multiple BrokerClasses reconciling a single Broker, so I'd prefer solving the smaller problem and only if we find it untenable with the actual use case we add that logic.

Second, I'd really like to use our MT Broker as the first implementor of this, and ideally it should not look any different from any other Broker. So, having some of this code in this file that shares the implementation seems non-ideal. I wonder if Set/GetConditionSet and the IsReady and InitializeConditions (maybe some other get toplevelcondition, or ??) should be the only ones here, and the Various Mark/Set methods should be moved to a new file, say: mt_broker_lifecycle.go? That way when a new Broker writer comes in it's easier to figure out what is implementation specific and what is not?
Would also make sure it will work for other Broker writers if we do NOT special case anything here. I think in particular I'm curious about calls to InitializeConditions? Perhaps there needs to be a default condSet so that still works?

Reply:
> My personal preference would be that we start with a single conditionSet that the controller sets to reduce complexity. As far as I understand there are no plans to have multiple BrokerClasses reconciling a single Broker, so I'd prefer solving the smaller problem and only if we find it untenable with the actual use case we add that logic.

Ok.

> Second, I'd really like to use our MT Broker as the first implementor of this, and ideally it should not look any different from any other Broker. So, having some of this code in this file that shares the implementation seems non-ideal. I wonder if Set/GetConditionSet and the IsReady and InitializeConditions (maybe some other get toplevelcondition, or ??) should be the only ones here, and the Various Mark/Set methods should be moved to a new file, say: mt_broker_lifecycle.go? That way when a new Broker writer comes in it's easier to figure out what is implementation specific and what is not?

Sure, agree. 
In particular, I wouldn't consider "the Various Mark/Set methods", methods of the `Broker` anymore but "helper" functions specific of our reconciler and not of the `Broker` object itself. 
In this way when a new Broker writer comes in they don't need to think that there is a method `Broker.Status.Mark.*Channel.*` that I cannot call. 
However, this requires lots of changes that for now, I consider out of scope for this PR. (If you agree I'll do a follow-up)
If you don't have objections, I would call the new file `broker_lifecycle_mt.go` to keep this file below `broker_lifecycle.go` in sorted filesystem viewers.

> Would also make sure it will work for other Broker writers if we do NOT special case anything here. I think in particular I'm curious about calls to InitializeConditions? Perhaps there needs to be a default condSet so that still works?

I would leave as default the "old" conditionSet.